### PR TITLE
[RNMobile] Fix media upload placeholder

### LINKED ISSUE #1544: Refine background/icon of image block transitional state
While reviewing [this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/12834), I noticed something that's been bugging me about the image block in general – the icon that sits over the background during upload progress is way too large and the wrong color. It looks like this:

image

It looks like the Video block has the "proper" styling, which IIRC is:

* Background: good as-is (same as global placeholder background, which IIRC is `gray-light` from [Colors Archive](https://dotcombrand.wordpress.com/colors-archive/) in `light mode` and `gray 90` from [Color Studio](https://color-studio.blog) in `dark mode`)
* Icon size: 32x32
* Icon color (`light mode`): `gray-lighten-20` from [Colors Archive](https://dotcombrand.wordpress.com/colors-archive/)
* Icon color (`dark mode`): `gray-70` from [Color Studio](https://color-studio.blog)

Which looks like this:

image

## Tasks

- [ ] Apply above styling to image block upload state placeholder
- [ ] Cross-reference other blocks that have this sort of "in between" state/background, to ensure we are matching the above styling?

### PR DESCRIPTION
**Pull request by cameronvoell**
## Description

Fixes [gutenberg-mobile #1544](https://github.com/wordpress-mobile/gutenberg-mobile/issues/1544)
This change refines the mobile image and video block placeholder shown during upload. 

## How has this been tested?

See gutenberg-mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1757
Uploading images can be tested using the WordPress Android and iOS apps using the following steps:

1. Add new Image or Video block
2. Click to upload a new image from wordpress media library or from device
3. Note that during upload we briefly see the improved placeholder.  

## Screenshots 
Placeholders before the change:
image

After Change
image

Video:
image-upload-gif

## Types of changes

This change is introducing deliberate, consistent styles for both light and dark mode for image and video placeholder icons and backgrounds during upload.  
There is also a small refactor for fetching icons in an easier to comprehend way and removing unreachable code. 

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. 
- [X] My code follows the accessibility standards. 
- [X] My code has proper inline documentation. 
- [X] I've included developer documentation if appropriate. 
- [X] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

File: packages/block-library/src/image/styles.native.scss [hunk 1]
```diff
@@ -31,22 +31,53 @@
...
+.iconUpload {
+	fill: $gray-lighten-20;
+	width: 100%;
+	height: 100%;
+}
+
+.iconUploadDark {
+	fill: $gray-70;
+	width: 100%;
+	height: 100%;
```
Comment (etoledom) [inline packages/block-library/src/image/styles.native.scss]: In the dark version of the style we don't need to re-define other than the color properties. All properties that are the same will be copied from the light version.

(Same for all other Dark style definitions)

**Review by etoledom — COMMENTED** [review]
Looks and works great! Thank you for this @cameronvoell 🎉 

Added a small code comment but it's just a small detail. 👍

File: packages/block-library/src/image/styles.native.scss (hunk 1 above)
Comment (cameronvoell) [reply packages/block-library/src/image/styles.native.scss]: Ah I see, thanks! Updated f32aab37dcdaaeeaa1ae16dd1e8b8a87bab8b001

**Review by etoledom — APPROVED** [review]
Thank you for the update @cameronvoell 
Great job! 🎉

**Pull request merged**